src/search/find-visits.js

```diff
@@ -64,13 +66,54 @@ export function getLastVisits({
 // XXX: If pages are redirected, only visits to the source page are found.
 export function findVisitsToPages({pagesResult}) {
     const pageIds = pagesResult.rows.map(row => row.id)
+   /**
+     * Here the whole data range values (StartDate and endDates) are accessed that are bieng updated 
+     * by Overview.jsx via date-picker . if they are not updated i.e user didn't seleceted any of them 
+     * then the startDate is intialized with default value of 100 days past and endDate is intialized 
+     * with present date.  if only one is selected other is initializedd with  it's default value.
+     * inside db,find we are  fetching only these data values that are between staertDate and endDate.
+     * Raj Pratim Bhattacharya gmail rajpratim1234@gmail.com
+     */
+    
+    var comp = new Date();
+    var sDate = comp.getTime() - 100*24*60*60*1000 //100 days old search
+    var eDate =  comp.getTime()
+    
+    if(ourState(store.getState()).nlp_date!='')
+        var result1 = parser.parseMoment( ourState(store.getState()).nlp_date );
```

Comment:
variable declaration inside if statement is not adviced (and inconsistent because of a gray area of the ES spec)